app/models/user.rb

```diff
@@ -1,2 +1,12 @@
+class EmailValidator < ActiveModel::EachValidator
+  def validate_each(record, attribute, value)
+    unless value =~ /\A([^@\s]+)@((?:[-a-z0-9]+\.)+[a-z]{2,})\z/i
+      record.errors.add attribute, options[:message] || "'#{value}' is not a valid email"
+    end
+  end
+end
```

Comment:
This class looks cool, but idk if the top of the `models/user.rb` is the right place for this.

Maybe we ask Mike about where we should put this?